paldb/src/main/java/com/linkedin/paldb/impl/StorageReader.java

```diff
@@ -350,7 +350,7 @@ public int getKeyCount() {
   }
 
   //Get data from disk
-  private byte[] getDiskBytes(long offset)
+  private synchronized byte[] getDiskBytes(long offset)
```

Comment:
Does this need to be synchronized?

RandomAccessFile is slow anyway.  It is synchronized.   SeekableFileChannel is faster, and a memory mapped file is faster than that.  Perhaps changing to one of those would be better.